### PR TITLE
repeat, repeat one, shuffle, no repeat, add permission

### PR DESCRIPTION
**Pull request by GHLong1997**

### CODE REVIEW

File: app/src/main/java/com/framgia/music/screen/playmusicscreen/PlayMusicFragment.java [hunk 1]
```diff
@@ -69,8 +70,10 @@
     private Utilities mUtilities;
     private PlayMusicService mPlayMusicService;
     private LinearLayout mLinearBottomParentActivity;
+    private String mSetup;
+    private boolean lastNonTrack;
 
-    public static PlayMusicFragment newInstance(Collection collection, int position) {
+    public static PlayMusicFragment newInstance(Collection collection, int position, boolean isLocalTrack) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/screen/playmusicscreen/PlayMusicFragment.java]: ERROR: Line is longer than 100 characters (found 108).

File: app/src/main/java/com/framgia/music/screen/playmusicscreen/PlayMusicFragment.java [hunk 2]
```diff
@@ -149,14 +152,21 @@ public void onClick(View view) {
...
             case R.id.image_play:
-                if (mPlayMusicService.isPlaying()) {
+                if (lastNonTrack) {
+                    mPlayMusicService.playTrack(mTrackIndex);
+                    mImageViewPlay.setImageDrawable(
+                            getResources().getDrawable(R.drawable.ic_pause_white_36dp));
+                    mImageViewArtParentActivity.setImageDrawable(
+                            getResources().getDrawable(R.drawable.ic_pause_black_36dp));
+                    updateProgressBar();
+                }else if (mPlayMusicService.isPlaying()) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/screen/playmusicscreen/PlayMusicFragment.java]: ERROR: 'else' is not preceded with whitespace.

File: app/src/main/java/com/framgia/music/screen/splashscreen/SplashScreenActivity.java [hunk 3]
```diff
@@ -72,4 +78,25 @@ private void handles() {
             finish();
         }, DELAY_2900);
     }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
+            @NonNull int[] grantResults) {
+        switch (requestCode) {
+            case Constant.REQUEST_WRITE_EXTERNAL_STORAGE: {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/screen/splashscreen/SplashScreenActivity.java]: WARNING: Avoid nested blocks.

File: app/src/main/java/com/framgia/music/service/PlayMusicService.java [hunk 4]
```diff
@@ -159,19 +165,91 @@ public void playTrack(int index) {
...
+        }
+
+        setupPreferences(isShuffle, isRepeat, isRepeatOne, isNonRepeat, mSetup);
+    }
+
+    public String getSetup() {
+        return mSetup;
+    }
+
+    private void setupPreferences(boolean isShuffle, boolean isRepeat, boolean isRepeatOne,
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/service/PlayMusicService.java]: WARNING: 'isShuffle' hides a field.

Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/service/PlayMusicService.java]: WARNING: 'isRepeat' hides a field.

Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/service/PlayMusicService.java]: WARNING: 'isRepeatOne' hides a field.

File: app/src/main/java/com/framgia/music/service/PlayMusicService.java [hunk 5]
```diff
@@ -159,19 +165,91 @@ public void playTrack(int index) {
...
+            boolean isNonRepeat, String mSetup) {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/service/PlayMusicService.java]: WARNING: 'isNonRepeat' hides a field.

Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/service/PlayMusicService.java]: WARNING: 'mSetup' hides a field.

File: app/src/main/java/com/framgia/music/utils/common/PermissionUtils.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import android.content.pm.PackageManager;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import com.framgia.music.utils.Constant;
+
+/**
+ * Created by Admin on 3/24/2018.
+ */
+
+public final class PermissionUtils {
```
Comment (le-quang-dao) [inline app/src/main/java/com/framgia/music/utils/common/PermissionUtils.java]: WARNING: Utility classes should not have a public or default constructor.

**Comment by daolq3012** [comment]
OK!

**Pull request merged**